cmake/extensions.cmake

```diff
@@ -397,8 +397,8 @@ function(zephyr_library_compile_options item)
   # zephyr_interface will be the first interface library that flags
   # are taken from.
 
-  string(RANDOM random)
-  set(lib_name options_interface_lib_${random})
+  string(MD5 uniqueness ${item})
+  set(lib_name options_interface_lib_${uniqueness})
```

Comment:
What if we add the same item twice?

Expected behaviour would for it to be silently added once due to CMake's de-duplication pass. But now I expect we will get an obscure error.

Perhaps we need to add our own "if not exists" check to silently de-duplicate.

Reply:
You're right, well-spotted and thx! I filed bug #15093 and submitted de-duplication fix #15094, thx for reviewing.

I also temporarily reverted this for testing and cmake is indeed de-duplicating identical compile options coming from multiple pseudo-libraries.